src/exporters/cds.py

```diff
@@ -0,0 +1,66 @@
+import cdsapi
+from pathlib import Path
+import warnings
+
+from typing import Dict
+
+from .base import BaseExporter
+
+
+class CDSExporter(BaseExporter):
+    """Exports for the Climate Data Store
+
+    cds.climate.copernicus.eu
+    """
+
+    def __init__(self, data_folder: Path = Path('data')) -> None:
+        super().__init__(data_folder)
+        self.client = cdsapi.Client()
+
+    @staticmethod
+    def make_filename(dataset: str, selection_request: Dict) -> str:
+        """Makes the appropriate filename for a CDS export
+        """
+        date_str = ''
+        if 'year' in selection_request:
+            years = selection_request['year']
+            if len(years) > 1:
```

Comment:
TODO: break up exports of more than a year and do them in chunks.

This will need to happen whether the argument is passed as `year` or `date`

Reply:
@gabrieltseng shall we use the [`chunks` function](https://github.com/tommylees112/esowc_notes/compare/vhi_preprocess#diff-6563edbe8608e8420b07391a54e0c862R85) in a `utils.py` file somewhere to chunk up the requests and send them separately?

But we can still use the [batch_ftp_request](https://github.com/tommylees112/esowc_notes/compare/vhi_preprocess#diff-6563edbe8608e8420b07391a54e0c862R59) as a template.

Or the [`download_file_from_ftp(ftp_instance, filename, output_filename)` function](https://github.com/tommylees112/esowc_notes/compare/vhi_preprocess#diff-6563edbe8608e8420b07391a54e0c862R45)


Actually we might not need it because the max number of clients we're going to spin up is the number of years, so max `40`. 

What do you think?

Reply:
where do we do this? This code is inside the `make_filename` function but do we not need to do this elsewhere?

Reply:
@gabrieltseng How do you use `date`? I haven't seen it in any of the API requests?

Reply:
yea, if we are going to do the `variable/year/month/file` as discussed this whole function will need to be totally rewritten.

`date_str` is used to make the filename, so that each file being downloaded is uniquely identified by its variable and date range